### DISCUSSION TITLE
CORRUPT HEAP: Bad head at 0x3ffe854c. Expected 0xabba1234 got 0xc5d1567c

### DISCUSSION THREAD

**Discussion by mln2007** [schreibfaul1/ESP32-audioI2S#933] [Q&A]
#include <WiFi.h>
#include <Audio.h>

#define I2S_DOUT      26  
#define I2S_BCLK      27  
#define I2S_LRC       25  
#define MAX_VOL       21  
#define VOL_UP        22  
#define VOL_DOWN      21  

Audio audio;

const char* ssid = "xxxx";
const char* pwd = "xxxx";

uint8_t volume = 10;    

void setup() {
  Serial.begin(115200);
  pinMode(VOL_UP, INPUT_PULLUP);
  pinMode(VOL_DOWN, INPUT_PULLUP);
  WiFi.mode(WIFI_STA);
  WiFi.begin(ssid, pwd);
  while (WiFi.status() != WL_CONNECTED) {
    Serial.print(".");
    delay(500);
  }
  Serial.println("connected to Wi-Fi");
  audio.setPinout(I2S_BCLK, I2S_LRC, I2S_DOUT);
  audio.setVolume(volume); // 0...21
  // ICRT電台
  // TED Talks
  //audio.connecttohost("https://delivery-cdn-cf.adswizz.com/audiohq_newEnco/ad_551941_2518a10745a1daa0b5873345178f8668_862c32a8026fd29ed4a35fc369cf2f23_91.mp3");
  // vocaloid FM
  audio.connecttohost("https://vocaloid.radioca.st/stream");
  //audio.connecttohost("https://stream1-epic-piano.radiohost.de/romantic-piano?ref=mytuner");
 //audio.connecttohost("http://fm901.cityfm.tw:8080/901.mp3");
 
}

void loop()
{
  audio.loop();

  if (digitalRead(VOL_DOWN) == LOW) {
    delay(20);
    if (digitalRead(VOL_DOWN) == LOW) {
      if (volume > 0) {
        volume--;
      }
      audio.setVolume(volume);
    }
  }

  if (digitalRead(VOL_UP) == LOW) {
    delay(20);
    if (digitalRead(VOL_DOWN) == LOW) {
      if (volume < MAX_VOL) {
        volume++;
      }
      audio.setVolume(volume);
    }
  }
}

// 事件處理函式
void audio_info(const char *info){
    printf("聲音資訊：%s\n", info);
}
void audio_showstation(const char *info){
    printf("電台   ：%s\n", info);
}
void audio_showstreaminfo(const char *info){
    printf("串流資訊：%s\n", info);
   
    
}
void audio_showstreamtitle(const char *info){
    printf("串流標題：%s\n", info);
}

This is my procedure and i connected to https://vocaloid.radioca.st/stream.When the next song played, an error occurs on the serial port.just like this:CORRUPT HEAP: Bad head at 0x3ffe854c. Expected 0xabba1234 got 0xc5d1567c

assert failed: multi_heap_free multi_heap_poisoning.c:276 (head != NULL)

Backtrace: 0x40082879:0x3ffb1f80 0x4008c6a9:0x3ffb1fa0 0x40091a76:0x3ffb1fc0 0x40090c01:0x3ffb20f0 0x400839bb:0x3ffb2110 0x40085849:0x3ffb2130 0x40158556:0x3ffb2150 0x40158582:0x3ffb2170 0x4014a0f0:0x3ffb2190 0x401009dc:0x3ffb21b0 0x400ffca5:0x3ffb21d0 0x400ffee9:0x3ffb21f0 0x400e260d:0x3ffb2210 0x400e359b:0x3ffb2230 0x400d3172:0x3ffb2250 0x401040f0:0x3ffb2270 0x4008edb6:0x3ffb2290

ELF file SHA256: c789ac99e05542a3

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4604
ho 0 tail 12 room 4
load:0x40078000,len:15488
load:0x40080400,len:4
load:0x40080404,len:3180
entry 0x400805b8
 
so i want to ask why and how to solve it.thx

**Comment by nstepanets** [comment]
It looks like there's a problem with the m_lastM3U8host variable.
